state/constraints.go

```diff
@@ -15,6 +15,11 @@ import (
 	"github.com/juju/juju/core/instance"
 )
 
+type constraintsWithID struct {
```

Comment:
This same type is implemented in _state/allwatcher.go_. We should just add `DocID` to `constraintsDoc` and remove usage of the composed types.

This has already been done on the develop branch.

Reply:
I've moved this type from the state/allwatcher.go package.

We can't just add the DocID to the constraintsDoc without additional changes, and I was trying to limit the churn on this branch.